src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/VS2kAddItemCommandHandler.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Input;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands
+{
+    [ExportCommandGroup(CommandGroup.VisualStudioStandard2k)]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal class VS2kAddItemCommandHandler : AbstractAddItemCommandHandler
+    {
+        protected override Dictionary<long, List<TemplateDetails>> GetTemplateDetails() => new Dictionary<long, List<TemplateDetails>>
+        {
+           // Command Id                                        Capability                      AdditionalCapability                DirNamePackageGuid          DirNameResourceId                                       TemplateNameResourceId
+            { VisualStudioStandard2KCommandId.AddForm,          ProjectCapability.CSharp,       ProjectCapability.WindowsForms,     LegacyCSharpPackageGuid,    LegacyCSharpStringResourceIds.IDS_PROJECTITEMTYPE_STR,  LegacyCSharpStringResourceIds.IDS_TEMPLATE_NEWWFCWIN32FORM  },
```

Comment:
TBH, I find the extension methods make it very hard to see what's going on here.

Reply:
In that I couldn't connect how this was related to the TemplateDetails.

Reply:
I may have misunderstood your request to make this into a table.

You prefer:
```
{ VisualStudioStandard2KCommandId.AddForm, new TemplateDetails(ProjectCapability.CSharp,       ProjectCapability.WindowsForms,     LegacyCSharpPackageGuid,    LegacyCSharpStringResourceIds.IDS_PROJECTITEMTYPE_STR,  LegacyCSharpStringResourceIds.IDS_TEMPLATE_NEWWFCWIN32FORM) }
```
?

Reply:
> I couldn't connect how this was related to the TemplateDetails.

Yes, I can see how that could be confusing. I may have gone too far with the feedback of just wanting to see the table of commands, and not have to worry about the implementation details.

Reply:
A static method to still simplify the add would help alleviate this confusion.

Reply:
Or constructor overload.